ql/src/java/org/apache/hadoop/hive/ql/Driver.java

```diff
@@ -536,6 +538,21 @@ public void run() {
       conf.set("mapreduce.workflow.id", "hive_" + queryId);
       conf.set("mapreduce.workflow.name", queryStr);
 
+      String timeStamp = Long.toString(System.currentTimeMillis());
+      int filterInd = queryStr.toLowerCase().indexOf("where");
+      String appId = "hive_" + timeStamp + queryStr.substring(0, Math.min(filterInd, 80))
```

Comment:
This line has a bunch of assumptions:
1. All SQL queries will have where clause in them. Not true
2. This code path will only be used by select queries. What happens if we have DDL statements? Did you test for that?
3. All sensitive information is before a where clause. Is this true? There is CASE clauses people can which appear before where. Somehow this seems hacky to me and needs to be fixed